### PR TITLE
Resource and Data Source Implementation for Security Policies.

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: examples/resources/bloxone_td_security_policy/resource.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+resource "bloxone_td_security_policy" "example" {
+  name = "example_security_policy"
+
+  # Other optional fields
+  rules = [
+    {
+      action = "action_allow",
+      data   = "terraform_example",
```
Comment (mathewab) [inline examples/resources/bloxone_td_security_policy/resource.tf:8]:
```suggestion
      data   = "custom_list_example",
```

File: examples/resources/bloxone_td_security_policy/resource.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+resource "bloxone_td_security_policy" "example" {
```
Comment (mathewab) [inline examples/resources/bloxone_td_security_policy/resource.tf:1]: Please add example for other optional fields too if you can

File: docs/data-sources/td_security_policies.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "bloxone_td_security_policies Data Source - terraform-provider-bloxone"
+subcategory: ""
```
Comment (mathewab) [inline docs/data-sources/td_security_policies.md:4]: Please ensure subcategory is "Threat Defense". You might have to modify the template here. https://github.com/infobloxopen/terraform-provider-bloxone/blob/master/templates/resources.md.tmpl

Comment (unasra) [reply docs/data-sources/td_security_policies.md:4]: I was planning to do this in a separate PR , once this is merged ,  as the doc file for all TD objects i.e. access codes , pop regions , Domain Lists. etc. would be changed.

File: examples/resources/bloxone_td_security_policy/resource.tf (hunk 1 above)
Comment (unasra) [reply examples/resources/bloxone_td_security_policy/resource.tf]: Resolved

File: go.mod [hunk 4]
```diff
@@ -4,11 +4,11 @@ go 1.19
 
 require (
 	github.com/hashicorp/terraform-plugin-docs v0.16.0
-	github.com/hashicorp/terraform-plugin-framework v1.4.2
+	github.com/hashicorp/terraform-plugin-framework v1.7.0
```
[the remark this answers is not in the document]
Comment (unasra) [reply go.mod]: Reverted back to 4.2

File: examples/resources/bloxone_td_security_policy/resource.tf (hunk 2 above)
Comment (unasra) [reply examples/resources/bloxone_td_security_policy/resource.tf:1]: Added a few fields for now , once DFP is merged , we can add that too

**Review by mathewab — APPROVED** [review]

**Pull request merged**